packages/languages/src/browser/language-client-contribution.ts

```diff
@@ -158,15 +167,21 @@ export abstract class BaseLanguageClientContribution implements LanguageClientCo
         return {
             documentSelector,
             synchronize: { fileEvents, configurationSection },
-            initializationFailedHandler: err => {
-                const detail = err instanceof Error ? `: ${err.message}` : '.';
-                this.messageService.error(`Failed to start ${this.name} language server${detail}`);
-                return false;
-            },
+            initializationFailedHandler: err => this.handleInitializationFailed(err),
             diagnosticCollectionName: id,
             initializationOptions
         };
     }
+    protected handleInitializationFailed(err: ResponseError<InitializeParams> | Error | any): boolean {
```

Comment:
[minor]: Why did you declare the union type? An `any` would be sufficient. I bet you had a good reason, I would like to know.

Reply:
I've copied typings from `vscode-languaclient` API. `any` is indeed sufficient for everything, but i find APIs giving a clue about what it can be are more helpful.